### PR TITLE
Fix replication metrics when using redis

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/replication/tcp/protocol.py [hunk 1]
```diff
@@ -86,6 +83,14 @@
     "synapse_replication_tcp_protocol_close_reason", "", ["reason_type"]
 )
 
+tcp_inbound_commands = Counter(
```
Comment (richvdh) [inline synapse/replication/tcp/protocol.py:86]: I wouldn't mind if these had a name which reflected that they were counters, now that they're becoming part of the public interface to this module. In other words: can you rename them to `tcp_*_commands_counter` ?

(I'll also remind you that the second field is meant to be a description, but happy if you want to leave that for now)

File: synapse/replication/tcp/protocol.py [hunk 2]
```diff
@@ -83,12 +83,16 @@
     "synapse_replication_tcp_protocol_close_reason", "", ["reason_type"]
 )
 
-tcp_inbound_commands = Counter(
-    "synapse_replication_tcp_protocol_inbound_commands", "", ["command", "name"],
+tcp_inbound_commands_counter = Counter(
+    "synapse_replication_tcp_protocol_inbound_commands",
+    "Number of commands received from replication",
```
Comment (richvdh) [inline synapse/replication/tcp/protocol.py:88]:
```suggestion
    "Number of commands received over replication, by command and connection ID",
```

File: synapse/replication/tcp/protocol.py [hunk 3]
```diff
@@ -83,12 +83,16 @@
...
+    ["command", "name"],
 )
 
-tcp_outbound_commands = Counter(
-    "synapse_replication_tcp_protocol_outbound_commands", "", ["command", "name"],
+tcp_outbound_commands_counter = Counter(
+    "synapse_replication_tcp_protocol_outbound_commands",
+    "Number of commands sent to replication",
```
Comment (richvdh) [inline synapse/replication/tcp/protocol.py:94]:
```suggestion
    "Number of commands sent out over replication, by command and connection ID",
```

File: synapse/replication/tcp/protocol.py (hunk 2 above)
Comment (erikjohnston) [reply synapse/replication/tcp/protocol.py:88]: They're not connection IDs, they're what they're connected to, will change

**Pull request merged**